### ISSUE TITLE
aws-cli/assume-role-with-web-identity sometimes fails

### ISSUE THREAD

**Issue by dejanceltra** [CircleCI-Public/aws-cli-orb#136]
We are using AWS CLI orb, and `assume-role-with-web-identity` command, which sometimes (rarely) fails with:
> An error occurred (InvalidIdentityToken) when calling the AssumeRoleWithWebIdentity operation: Couldn't retrieve verification key from your identity provider,  please reference AssumeRoleWithWebIdentity documentation for requirements
Failed to assume role

CircleCI support redirected me to the following two urls:
- https://repost.aws/knowledge-center/iam-sts-invalididentitytoken
- https://github.com/aws/aws-cli/issues/7194

However, this seems like a functionality that should be implemented in this orb, to ensure that it's fixed for all consumers.

**Comment by davidvasandani** [comment]
We're testing a [fix](https://github.com/CircleCI-Public/aws-cli-orb/pull/137) that adds a retry and sleep. While this should be enough, we're open to suggestions, as I can imagine there is a better solution.

**Comment by RainofTerra** [comment]
oh and, could this be backported to a 3.x release? We haven't moved to the 4.x release yet and I know there are breaking changes.

**Comment by davidvasandani** [comment]
Regarding a 3.x release: Yes! We haven't moved either. I've updated the PR from v3.1.5 and pushed a public Orb that we'll be testing tomorrow `sweetgreen/aws-cli@3.1.7`

**Comment by brivu** [comment]
Hey Everyone (@davidvasandani, @RainofTerra, @dejanceltra),

I'm Brian from the Community and Partner team here at CircleCI. I'm working on this issue at the moment am planning on enabling standard retries directly from the `aws cli` by adding the following to the `~/.aws/config` file: 

```
[default]
retry_mode = standard
max_attempts = 6
```

https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-retries.html#cli-usage-retries-modes-standard.title

I am having trouble testing this solution because I cannot reproduce the issue with the `aws-cli/assume-role-with-web-identity` command. 

If anyone can give this a shot for me and let me know if it works, I'd really appreciate it!

Thank you for all of your contributions!

Best, 
Brian

**Comment by dnguy078** [comment]
I've tried this solution by adding in 

```
export AWS_RETRY_MODE=standard
export AWS_MAX_ATTEMPTS=6
```

This solution did not seem to work yesterday. 
In any case, I've seen this issue go away today. It's possible AWS may have made some changes on their side regarding this endpoint.

**Comment by brivu** [comment]
Hey @dnguy078, 

Thanks for trying. I think I am going to merge @davidvasandani's PR #137 real soon. 

Best, 
Brian

**Comment by brivu** [comment]
Hey Everyone, 

I've merged PR #137 and back ported it to version `3.2`. It's been released.

I'm going to close this issue for now. 

Thanks! 
Brian
